src/main/java/com/yelp/nrtsearch/server/query/QueryNodeMapper.java

```diff
@@ -580,11 +576,15 @@ private static Query getPrefixQuery(PrefixQuery prefixQuery, IndexState state) {
       throw new IllegalArgumentException(
           "Field \"" + prefixQuery.getField() + "\" is not indexed with terms");
     }
```

Comment:
You can get rid of the checks above, as they are not needed when using the new interface. Any needed checks can be done in the interface implementation